lucene/core/src/java/org/apache/lucene/search/FeatureComparatorSource.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.lucene.search;
+
+import java.io.IOException;
+
+import org.apache.lucene.document.FeatureField;
+import org.apache.lucene.index.LeafReaderContext;
+import org.apache.lucene.index.PostingsEnum;
+import org.apache.lucene.index.Terms;
+import org.apache.lucene.index.TermsEnum;
+import org.apache.lucene.util.BytesRef;
+
+/** Provides a comparator that parses a feature field's values as float and sorts by descending value */
+public class FeatureComparatorSource extends FieldComparatorSource {
+
+  private final BytesRef featureName;
+  private final Float missingValue;
+
+  /**
+   * Creates a {@link FieldComparatorSource} that can be used to sort hits by
+   * the value of a particular feature in a {@link FeatureField}. Note that the
+   * field name to use for the sort will be specified by the {@link SortField}.
+   *
+   * @param featureName The name of the feature to use for the sort value
+   * @param missingValue The value to use for documents that don't have a value
+   *   for the feature. If <code>null</code>, documents with no value for the
+   *   feature will be sorted as if their value is <code>0.0f</code>.
+   */
+  public FeatureComparatorSource(BytesRef featureName, Float missingValue) {
+    this.featureName = featureName;
+    this.missingValue = missingValue;
```

Comment:
FeatureField is mostly useful for queries, where we don't support default values. I don't think supporting default values at sort time is going to be useful.

Reply:
So we should always assume a value of `0.0f` if the value is missing then? I think that makes sense since the feature is stored as a term frequency so not having a value corresponds to having a term frequency of 0

Reply:
Yes, this was my reasoning indeed.